### ISSUE TITLE
Error -4090 EADDRNOTAVAIL address not available

### ISSUE THREAD

**Issue by mmedic** [aspnet/KestrelHttpServer#1836]
I have a .net core web api project created and deployed it to Azure. When the app starts it is throwing this error? None of the request are being processed.

[WRN] Unable to bind to http://localhost:22676 on the IPv6 loopback interface. (de7f78a0)
System.AggregateException: One or more errors occurred. (Error -4090 EADDRNOTAVAIL address not available) ---> Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -4090 EADDRNOTAVAIL address not available
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.tcp_bind(UvTcpHandle handle, SockAddr& addr, Int32 flags)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvTcpHandle.Bind(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.TcpListener.CreateListenSocket()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Listener.<StartAsync>b__8_0(Object state)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.KestrelEngine.CreateServer(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.KestrelServer.Start[TContext](IHttpApplication`1 application)
---> (Inner Exception #0) Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -4090 EADDRNOTAVAIL address not available
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.tcp_bind(UvTcpHandle handle, SockAddr& addr, Int32 flags)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvTcpHandle.Bind(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.TcpListener.CreateListenSocket()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Listener.<StartAsync>b__8_0(Object state)<---

**Comment by davidfowl** [comment]
What version of Kestrel are you running? That error is expected and it's just a warning and likely has nothing to do with the issue you're seeing.

**Comment by mmedic** [comment]
Here is my reference:

<PackageReference Include="Microsoft.AspNetCore.Server.Kestrel" Version="1.1.1" />

**Comment by davidfowl** [comment]
Are you getting an error? What does "None of the request are being processed." mean? Do you have any other logs?

**Comment by mmedic** [comment]
I mean that using any REST client, hitting a wide open endpoint results in a timeout after 30 seconds. 

Logs don't show much more. After app initializes (i.e. connects to the database, runs migrations etc.) logging starts:

```
> 2017-05-13T19:50:30.4193676+00:00  [INF] Azure Web Sites environment detected. Using '"D:\home\ASP.NET\DataProtection-Keys"' as key repository; keys will not be encrypted at rest. (383ad3af)
> 2017-05-13T19:50:30.4506078+00:00  [WRN] Disabling the transport security requirement is not recommended in production. Consider setting 'OpenIdConnectServerOptions.AllowInsecureHttp' to 'false' to prevent the OpenID Connect server middleware from serving non-HTTPS requests. (4405708c)
> 2017-05-13T19:50:30.7318677+00:00  [WRN] Unable to bind to http://localhost:22992 on the IPv6 loopback interface: (Error -4090 EADDRNOTAVAIL address not available) (74dd9af9)
> 2017-05-13T19:50:31.1693711+00:00 0HL4Q757VKDCG [INF] Request starting HTTP/1.1 GET http://icbnc.azurewebsites.net/   (e5be5b71)
> 2017-05-13T19:50:31.9662428+00:00 0HL4Q757VKDCG [INF] Request finished in 845.2043ms 200 text/html; charset=utf-8 (15c52c40)
> 2017-05-13T19:50:32.1849866+00:00 0HL4Q757VKDCH [INF] Request starting HTTP/1.1 GET http://icbnc.azurewebsites.net/favicon.ico   (e5be5b71)
> 2017-05-13T19:50:32.1849866+00:00 0HL4Q757VKDCH [INF] Request finished in 1.0331ms 200 text/html; charset=utf-8 (15c52c40)
> 2017-05-13T19:50:40.3425427+00:00 0HL4Q757VKDCI [INF] Request starting HTTP/1.1 GET http://icbnc.azurewebsites.net/api/ping   (e5be5b71)
> 2017-05-13T19:50:40.4206742+00:00 0HL4Q757VKDCI [INF] Executing action method "Icbnc.WebApi.Controllers.PingController.Ping (Icbnc.WebApi)" with arguments (null) - ModelState is Valid (ba7f4ac2)
> 2017-05-13T19:50:40.4519379+00:00 0HL4Q757VKDCI [INF] Executing ObjectResult, writing value "Microsoft.AspNetCore.Mvc.ControllerContext". (4e968210)
> 2017-05-13T19:50:40.4675212+00:00 0HL4Q757VKDCI [INF] Executed action "Icbnc.WebApi.Controllers.PingController.Ping (Icbnc.WebApi)" in 83.5793ms (afa2e885)
> 2017-05-13T19:50:40.4675212+00:00 0HL4Q757VKDCI [INF] Request finished in 124.074ms 200 text/html; charset=utf-8 (15c52c40)
> 2017-05-13T19:50:40.5769097+00:00 0HL4Q757VKDCJ [INF] Request starting HTTP/1.1 GET http://icbnc.azurewebsites.net/favicon.ico   (e5be5b71)
> 2017-05-13T19:50:40.5769097+00:00 0HL4Q757VKDCJ [INF] Request finished in 0.3713ms 200 text/html; charset=utf-8 (15c52c40)
> 2017-05-13T19:50:49.1967789+00:00 0HL4Q757VKDCK [INF] Request starting HTTP/1.1 GET http://icbnc.azurewebsites.net/api/members/1   (e5be5b71)
> 2017-05-13T19:50:49.1967789+00:00 0HL4Q757VKDCK [INF] Request finished in 0.3533ms 200 text/html; charset=utf-8 (15c52c40)
> 2017-05-13T19:50:49.3061509+00:00 0HL4Q757VKDCL [INF] Request starting HTTP/1.1 GET http://icbnc.azurewebsites.net/favicon.ico   (e5be5b71)
> 2017-05-13T19:50:49.3061509+00:00 0HL4Q757VKDCL [INF] Request finished in 0.3791ms 200 text/html; charset=utf-8 (15c52c40)
> 
```

Even log seems to indicate that the process has started:
`<Events><Event><System><Provider Name="IIS AspNetCore Module"/><EventID>1004</EventID><Level>0</Level><Task>0</Task><Keywords>Keywords</Keywords><TimeCreated SystemTime="2017-05-13T19:50:24Z"/><EventRecordID>947087359</EventRecordID><Channel>Application</Channel><Computer>RD00155D8A4915</Computer><Security/></System><EventData><Data>Warning: Could not create stdoutLogFile \\?\D:\home\site\wwwroot\logs\stdout_14404_2017513195024.log, ErrorCode = -2147024893.</Data></EventData></Event><Event><System><Provider Name="IIS AspNetCore Module"/><EventID>1001</EventID><Level>0</Level><Task>0</Task><Keywords>Keywords</Keywords><TimeCreated SystemTime="2017-05-13T19:50:30Z"/><EventRecordID>947094015</EventRecordID><Channel>Application</Channel><Computer>RD00155D8A4915</Computer><Security/></System><EventData><Data>Application 'MACHINE/WEBROOT/APPHOST/ICBNC' started process '22488' successfully and is listening on port '22992'.</Data></EventData></Event></Events>`

**Comment by mmedic** [comment]
This turned to be an issue with my firewall.

**Comment by natemcmaster** [comment]
This looks like a duplicate of https://github.com/aspnet/KestrelHttpServer/issues/1001. Can you try Kestrel 1.1.2? The warning may still appear, but the stack trace will be suppressed. (See https://github.com/aspnet/KestrelHttpServer/pull/1677).

**Comment by mmedic** [comment]
Indeed, 1.1.2 shows the warning without the stack trace.

**Comment by natemcmaster** [comment]
Ok, it sounds like your issue is resolved, so closing as a duplicate of #1001.
